exercises/tree-building/examples/success-standard/src/TreeBuilding.hs

```diff
@@ -0,0 +1,77 @@
+module TreeBuilding (newTree, Record(..), Tree(..)) where
+
+import Data.List
+import Data.Maybe (isNothing)
+
+type Id = Int
+type Children = [Tree]
+type ParentGrouping = (Maybe Id, [Id])
+
+data Record = Record Id (Maybe Id) deriving (Eq, Show)
+
+data Tree = Leaf Id | Branch Id Children deriving (Eq, Show)
+
+newTree :: [Record] -> Maybe Tree
+newTree records
+  | cycles records      = Nothing
+  | succIdCheck records = build . groupByParent $ records
```

Comment:
to aid the understandability of `succIdCheck` in this context, I suggest the function be renamed to reflect what it signifies if it were to return true. For example, if I read it correctly, it seems to be `containsConsecutiveIds` ?